### DISCUSSION TITLE
Elasticsearch is unhealthy after uploading vcf

### DISCUSSION THREAD

**Discussion by GeorginaAmbriz** [broadinstitute/seqr#3803] [General]
Hello,
I made a local install of Seqr and I succeed in uploading the reference data and a vcf file with "Option #2: annotate and load on-prem". This means I made all points of local install and upload instructions from 1 to 5. I came back today to checkup on seqr installation, for this I made
`docker compose ps`
and I saw that elastic search was "unhealthy". I updated seqr with: 
```
docker-compose pull
docker-compose up -d seqr
```
But I had this error message:
`dependency failed to start: container elasticsearch is unhealthy`

seqr didn't restart and I had this output:
```
docker compose up -d seqr
[+] Running 3/3
 ✔ Container geo-redis-1     Healthy                                                                                                                                    0.0s 
 ✔ Container geo-postgres-1  Healthy                                                                                                                                    0.0s 
 ✘ Container elasticsearch   Error                                                                                                                                      0.0s 
dependency failed to start: container elasticsearch is unhealthy
```
Please find attached  the log of elasticsearch that I got with: 
`docker compose logs -f elasticsearch > elastic_search_logs.txt`

[elastic_search_logs.txt](https://github.com/broadinstitute/seqr/files/13834794/elastic_search_logs.txt)

I really appreciate any help you can provide about this issue,

Georgina Jimenez Ambriz
Bioinformatics Core
Neuro

**Comment by hanars** [comment]
try starting the services sequenctially and see if that fxes the issue:
```
docker compose down seqr
docker compose up elasticsearch
docker compose up seqr
```

**Comment by GeorginaAmbriz** [comment]
Hello,
Thank you for your answer. I ran the commands that you gave me, this is the output:
```
docker compose down seqr
[+] Running 2/0
 ✔ Container geo-seqr-1  Removed                                                                    0.0s 
 ! Network geo_default   Resource is still in use   
``` 
```
docker compose up elasticsearch
[+] Running 1/0
 ✔ Container elasticsearch  Running                                                                 0.0s 
Attaching to elasticsearch
```
The last command has been running for tow hour and it's still running. Is it normal? Can it take a lot of time?

**Reply by hanars** [reply]
yes sorry if that wasn't clear, but that command just runs elasticsearch indefinitely. You need to start seqr in another terminal

**Reply by GeorginaAmbriz** [reply]
Well, I ran the second command in  a different terminal but I still have the same problem:

```
docker compose up seqr
[+] Running 4/4
 ✔ Container elasticsearch   Running                                                                                                                                               0.0s 
 ✔ Container geo-postgres-1  Running                                                                                                                                               0.0s 
 ✔ Container geo-redis-1     Running                                                                                                                                               0.0s 
 ✔ Container geo-seqr-1      Created                                                                                                                                               0.6s 
Attaching to geo-seqr-1
dependency failed to start: container elasticsearch is unhealthy
```
I've tried restarting seqr several times but I got the same result with the same logs for elasticsearch. If I do `docker-compose logs -f seqr`, nothing happens, I don't have an output.

Any idea of what's going on?

**Reply by hanars** [reply]
I really need the elasticsearch logs, not the seqr logs, as elasticsearch is what is unhealthy

**Reply by GeorginaAmbriz** [reply]
hi,
Here are the elasticsearch logs.
Thank you
[elasticsearch_logs.txt](https://github.com/broadinstitute/seqr/files/13863216/elasticsearch_logs.txt)

**Reply by hanars** [reply]
What happens if you run `curl -s elasticsearch:9200/_cat/health`

**Reply by GeorginaAmbriz** [reply]
Nothing happens. I don't have an output after running `curl -s elasticsearch:9200/_cat/health`

**Reply by hanars** [reply]
what about `curl -s elasticsearch:9200/status`

**Reply by GeorginaAmbriz** [reply]
nothing happened, I didn't have an output this time neither.

**Reply by hanars** [reply]
Okay,  can you try fully restarting seqr and then running `docker-compose ps` and send me the output

**Reply by GeorginaAmbriz** [reply]
Hi,
When I try to fully restart seqr I have this output:

```
docker compose up -d seqr
[+] Running 3/3
 ✔ Container geo-redis-1     Healthy                                                                                                                         0.0s 
 ✘ Container elasticsearch   Error                                                                                                                           0.0s 
 ✔ Container geo-postgres-1  Healthy                                                                                                                         0.0s 
dependency failed to start: container elasticsearch is unhealthy
```
And this is the output of `docker compose ps`:
```
NAME             IMAGE                                                  COMMAND                                                        SERVICE         CREATED        STATUS                    PORTS
elasticsearch    docker.elastic.co/elasticsearch/elasticsearch:7.16.3   "/bin/tini -- /usr/local/bin/docker-entrypoint.sh eswrapper"   elasticsearch   26 hours ago   Up 26 hours (unhealthy)   0.0.0.0:9200->9200/tcp, :::9200->9200/tcp, 9300/tcp
geo-postgres-1   postgres:12.6                                          "docker-entrypoint.sh postgres -c listen_addresses=*"          postgres        4 days ago     Up 4 days (healthy)       5432/tcp
geo-redis-1      redis:6.2.7                                            "docker-entrypoint.sh redis-server"                            redis           4 days ago     Up 4 days (healthy)       6379/tcp
```

**Reply by hanars** [reply]
so a full restart should not have your elasticsearch running for 26 hours and redis up for 4 days. Please fully take down all the components and restart from scratch
